### DISCUSSION TITLE
0.8.4 Beta release

### DISCUSSION THREAD

**Discussion by Timthreetwelve** [Timthreetwelve/TimVer#20] [Announcements]
## This is a beta (pre-release) version of TimVer
**Please report any bugs by opening a [new Issue](https://github.com/Timthreetwelve/TimVer/issues/new/choose) or by commenting on an [existing issue](https://github.com/Timthreetwelve/TimVer/issues).**

### Changes:
- New: TimVer is now multilingual
- New: Updated UI
- New: Check for new releases from the About page
- New: Disk Drive Information page
- New: Graphics (Video) Information page
- New: Environment Variables page
- New: Processor description on the Hardware Information page
- New: BIOS information on the Hardware Information page
- Updated: Computer Information page has been renamed to Hardware Information
- Updated: Build history is now optional
- Updated: History file format changed to JSON
- Updated: Darker theme
- Updated: Memory display now includes installed and usable
- Updated: Added content to the [Wiki](https://github.com/Timthreetwelve/TimVer/wiki)

### Breaking change ⚡
> [!important]
> Build history is now optional, and the file format has changed. If you wish to keep your existing build history, you will need to download and run the history conversion program. 
<Details>
<Summary><h3>Steps to convert history</h3></Summary>

  - Find the `history.csv` file in the current TimVer folder. 
    - This is usually located in `C:\Users\<your user>\AppData\Local\Programs\T_K\TimVer`.
  - Copy the `history.csv` to another location, for example your desktop.
  - Install the latest version of TimVer.
  - Download ConvertHistory_0.1.0_Portable.zip which can be found in the Assets section below.
  - Copy the folder contained in the zip archive to a location of your choice.
  - Run ConvertHistory.exe.
  - Enter the full path to the `history.csv` file in the box labeled `Input CSV File`.
  - Enter the full path for the output JSON file in the box labeled `Output JSON File` or accept the default.
  - Click the `Convert` button.
  - If the conversion was successful, copy the `history.json` file to the TimVer folder.
  - Start TimVer and verify the build history.
  - If everything looks good, you can delete the `history.csv` file and the `ConvertHistory` folder.
</Details>

### New download and installation options ✨
With this release I'm adding *Self Contained* versions both the portable and installer versions. In this context, Self-Contained means that the required .NET files are included. A self-contained installation does **not** require that the .NET runtime be installed. Of course, by including the required .NET files, the installation is much larger. Download the installer or zip archive that best suits your needs.

In addition:
* Publishing a Self-Contained build requires that either x86 or x64 be targeted. Consequently, there are versions for each x86 and x64. 
* The filename of the Self-Contained builds will indicate the architecture, either `x86` or `x64`.
* The name of the zip archives has been changed from `NonInstall` has been changed to `Portable`.
* The **Checksums** table below has been updated to include a description for each file.

### Translations ✍🏻
My thanks to all the people that have contributed translations for TimVer. You rock!

If you wish to contribute a translation, check out the [Contribute a Translation](https://github.com/Timthreetwelve/TimVerP/wiki/Contribute-a-Translation) topic in the Wiki. Your help would be awesome.

#### Checksums :
| File Name | Description | SHA256 Hash |
| --- | --- | --- |
|ConvertHistory_0.1.0_Portable.zip|I-RN6-xEither|` 3349cc2713a76cf2cc3c035fd8e3bfceb41ddc2ba3f5becc4761bfa78ec603b8 `| 
|TimVer_0.8.4.0_beta_Setup.exe|I-RN6-xEither|` 390d324d5e46dcc0c2745aa9216e87a2112ea301765963488d6d001bfa2ed607 `|
|TimVer_0.8.4.0_beta_SC_x64_Setup.exe|I-SC-x86|` e89147a22f66d9c1e94825798719c9db5b05098be291dd60c5d328e9ed48843a `|
|TimVer_0.8.4.0_beta_SC_x86_Setup.exe|I-SC-x86|` c48e7c9dd5c93dfe9ff6c86dd4e6e35f9e5f4ae4f6969db16b488033a1b788c4 `|
|TimVer_0.8.4.0_beta_Portable.zip|Z-RN6-xEither|` 53f7dcd5e351bd1f8b44bd75f1b27a7ae2d7eb53a6ae73ba550ff9caebe9e833 `|
|TimVer_0.8.4.0_beta_SC_x64_Portable.zip|Z-SC-x64|` 5d40579bd4d6d67daa9ffe35d59f14ac542b17445334262cc9ca575b4c0b1713 `| 
|TimVer_0.8.4.0_beta_SC_x86_Portable.zip|Z-SC-x86|` 7702b5ec2f2e073abf8900d0aa0ecfb59f87115c9b044a4c1b9ecc0eb48cca47 `|

**Description key:**
- I = exe installer
- Z = zip archive
- RN6 = Requires existing .NET 6 runtimes
- SC = Self-Contained, existing .NET 6 runtimes are not required
- x64 = 64-bit architecture
- x86 = x86 architecture
- xEither = Either x64 or x86

**Full Changelog**: https://github.com/Timthreetwelve/TimVer/compare/v0.7.3...v0.8.4-beta.123

<hr /><em>This discussion was created from the release <a href='https://github.com/Timthreetwelve/TimVer/releases/tag/v0.8.4-beta.123'>0.8.4 Beta release</a>.</em>

**Comment by Timthreetwelve** [comment]
@bovirus @CMTriX

I released a beta version of TimVer a few minutes ago. Give it a spin.

Windows Defender initially reported the installer had a virus, but after updating definitions the warning disappeared.

Happy New Year 🎉

Tim

**Comment by CMTriX** [comment]
I am going to test it! TH@nx!
I am on the **oliebollen** now!
Happy New Year 🎉

**Reply by Timthreetwelve** [reply]
Thanks for testing. I've started and stopped working on TimVer so many times I've probably overlooked a thing or two.

Oliebollen looks awesome. I gained some weight just looking at pictures.
